### PR TITLE
Feature/thread creation 

### PR DESCRIPTION
**Pull request by DikenMaharjan**
Contains methods to retrieve and create threads.

### CODE REVIEW

File: src/test/kotlin/com/safehill/kclient/api/SafehillApiImplTests.kt [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+    private suspend fun createUserOnServer(
+        coroutineScope: CoroutineScope,
+        user: SHLocalUser? = null
+    ): SHLocalUser {
         val localUser: SHLocalUser = user ?: run {
             val cryptoUser = SHLocalCryptoUser()
             SHLocalUser(cryptoUser)
         }
 
-        val charPool : List<Char> = ('a'..'z') + ('A'..'Z') + ('0'..'9')
```
Comment (gennaro-safehill) [inline src/test/kotlin/com/safehill/kclient/api/SafehillApiImplTests.kt:30]: Do you mind defining carving out username validation and define a new method for it?

File: src/main/kotlin/com/safehill/kclient/api/dtos/SHRecipientEncryptionDetailsDTO.kt [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+
+@Serializable
 data class SHRecipientEncryptionDetailsDTO(
-    val userIdentifier: String,
+    val recipientUserIdentifier: String,
     val ephemeralPublicKey: String, // base64EncodedData with the ephemeral public part of the key used for the encryption
     val encryptedSecret: String, // base64EncodedData with the secret to decrypt the encrypted content in this group for this user
-    val secretPublicSignature: String // base64EncodedData with the public signature of the user sending it
+    val secretPublicSignature: String, // base64EncodedData with the public signature of the user sending it
+    val senderPublicSignature: String
```
Comment (gennaro-safehill) [inline src/main/kotlin/com/safehill/kclient/api/dtos/SHRecipientEncryptionDetailsDTO.kt]: Add documentation for it (you'll find it in the Swift counterpart)

**Review by gennaro-safehill — APPROVED** [review]
Looks great, just two minor things

File: src/main/kotlin/com/safehill/kclient/api/dtos/SHRecipientEncryptionDetailsDTO.kt (hunk 2 above)
Comment (DikenMaharjan) [reply src/main/kotlin/com/safehill/kclient/api/dtos/SHRecipientEncryptionDetailsDTO.kt]: Sure

File: src/test/kotlin/com/safehill/kclient/api/SafehillApiImplTests.kt (hunk 1 above)
Comment (DikenMaharjan) [reply src/test/kotlin/com/safehill/kclient/api/SafehillApiImplTests.kt:30]: I am not sure what you mean. Could you elaborate a bit? Do you want me to extract out the function that returns username?

**Pull request merged**